coderd/fositeprovider/fositestorage/storage.go

```diff
@@ -0,0 +1,72 @@
+package fositestorage
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/ory/fosite"
+	"github.com/ory/fosite/handler/oauth2"
+	"github.com/ory/fosite/storage"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/dbauthz"
+)
+
+type fositeStorage interface {
+	fosite.ClientManager
+	oauth2.CoreStorage
+}
+
+var _ fositeStorage = (*Storage)(nil)
+
+type Storage struct {
+	db database.Store
+
+	// TODO: Remove the memory store entirely and implement all methods to use the database.
+	storage.MemoryStore
```

Comment:
I think this is a good use-case for our idea of a separate `OAuthStore` as opposed to just throwing in a `database.Store`.